service/src/main/java/skills/services/events/SkillEventPublisher.groovy

```diff
@@ -19,16 +19,30 @@ import groovy.util.logging.Slf4j
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.messaging.simp.SimpMessagingTemplate
 import org.springframework.stereotype.Component
+import skills.websocket.SubscribedDestinationRegistry
 
 @Component
 @Slf4j
 class SkillEventPublisher {
 
+    @Autowired
+    SubscribedDestinationRegistry destinationRegistry
+
     @Autowired
     final SimpMessagingTemplate messagingTemplate;
 
     void publishSkillUpdate(SkillEventResult result, String userId) {
         log.debug("Reporting user skill for user [$userId], result [$result]")
-        messagingTemplate.convertAndSendToUser(userId, "/queue/${result.projectId}-skill-updates", result)
+        if(result.projectId) {
+            messagingTemplate.convertAndSendToUser(userId, "/queue/${result.projectId}-skill-updates", result)
+        } else {
+            List<String> destinations = destinationRegistry.getAllDestinationsForUser(userId)
+            log.debug("got [${destinations?.size()}] subscribed destinations for user [$userId]")
```

Comment:
remove string concatenation or wrap with isDebugEnabled check; looks like debug above also has that issue and although not part of the ticket would be nice to improve